File: db/java/app/src/main/java/com/example/sales/domain/model/CreditBalance.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 与信残高データのEntityクラス
 */
public class CreditBalance {
    private String companyCode;
    private Integer orderBalance;
    private Integer receivableBalance;
    private Integer payableBalance;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public CreditBalance() {}

    // Getter/Setter（全フィールド）
    public String getCompanyCode() { return companyCode; }
    public void setCompanyCode(String companyCode) {
        this.companyCode = companyCode;
    }

    public Integer getOrderBalance() { return orderBalance; }
    public void setOrderBalance(Integer orderBalance) {
        this.orderBalance = orderBalance;
    }

    public Integer getReceivableBalance() { return receivableBalance; }
    public void setReceivableBalance(Integer receivableBalance) {
        this.receivableBalance = receivableBalance;
    }

    public Integer getPayableBalance() { return payableBalance; }
    public void setPayableBalance(Integer payableBalance) {
        this.payableBalance = payableBalance;
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) {
        this.updatedBy = updatedBy;
    }
}

File: db/go/sales_management/internal/model/credit_balance.go
package model

import (
	"time"
)

// CreditBalance 与信残高のドメインモデル
type CreditBalance struct {
	CompanyCode        string    `db:"取引先コード"`
	OrderBalance       int       `db:"受注残高"`
	AccountsReceivable int       `db:"売掛残高"`
	AccountsPayable    int       `db:"買掛残高"`
	CreatedAt          time.Time `db:"作成日時"`
	CreatedBy          string    `db:"作成者名"`
	UpdatedAt          time.Time `db:"更新日時"`
	UpdatedBy          string    `db:"更新者名"`
}
